File: LC530.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {

private:
    int m = INT_MAX;
    int previous = INT_MAX;
public:

    void minTree(TreeNode *root){
        if(root == NULL)
            return;
        
        if(root->left)
            minTree(root -> left);

        m = min(m, abs(previous - root -> val));
        previous = root -> val;
        
        if(root->right)
            minTree(root -> right);
    }

    int getMinimumDifference(TreeNode* root) {
        minTree(root);
        return m;
    }
};

File: LC530.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:

    def getMin(self, root):
        if root == None:
            return
        
        if(root.left):
            self.getMin(root.left)
        
        self.m = min(self.m , abs(self.prev - root.val))
        self.prev = root.val

        if(root.right):
            self.getMin(root.right)

    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        self.m = float('inf')
        self.prev = float('inf')

        self.getMin(root)
        return self.m
